src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+        if( !aggregate.isEmpty()) {
+            reduceSum.addAll( addOverlappingVariants(aggregate, false, featureContext) );
+        }
+    }
+
+    /**
+     * Using the RefMetaDataTracker find overlapping variants and pull out the necessary information to create the VariantDatum
+     * @param featureInputs the input sources to search within
+     * @param isInput   is this rod an -input rod?
+     * @param context   the AlignmentContext from the RODWalker map call
+     * @return  a list of VariantDatums, can be empty
+     */
+    private List<VariantDatum> addOverlappingVariants(
+            final List<FeatureInput<VariantContext>> featureInputs,
+            final boolean isInput,
+            final FeatureContext context ) {
+        if( featureInputs == null ) { throw new IllegalArgumentException("featureInputs cannot be null."); }
+        if( context == null ) { throw new IllegalArgumentException("context cannot be null."); }
+
+        final ExpandingArrayList<VariantDatum> variants = new ExpandingArrayList<>();
+
+        for( final VariantContext vc : context.getValues(featureInputs, context.getInterval().getStart()) ) {
+            if( vc != null && ( IGNORE_ALL_FILTERS || vc.isNotFiltered() || ignoreInputFilterSet.containsAll(vc.getFilters()) ) ) {
+                if( VariantDataManager.checkVariationClass( vc, VRAC.MODE ) && !VRAC.useASannotations) {
+                    addDatum(variants, isInput, context, vc, null, null);
+                }
+                else if(VRAC.useASannotations) {
+                    for(final Allele allele : vc.getAlternateAlleles()) {
+                        if(allele == Allele.SPAN_DEL)
+                            continue;
+                        if (VariantDataManager.checkVariationClass(vc, allele, VRAC.MODE))
+                            addDatum(variants, isInput, context, vc, vc.getReference(), allele);
+                    }
+                }
+            }
+        }
+
+        return variants;
+    }
+
+    /**
+     * add a datum representing a variant site (or allele) to the data in {@code variants}, which represents the callset to be recalibrated
+     * @param variants is modified by having a new VariantDatum added to it
+     */
+    private void addDatum(
+            final ExpandingArrayList<VariantDatum> variants,
+            final boolean isInput,
+            final FeatureContext featureContext,
+            final VariantContext vc,
+            final Allele refAllele,
+            final Allele altAllele) {
+        final VariantDatum datum = new VariantDatum();
+
+        // Populate the datum with lots of fields from the VariantContext, unfortunately the VC is too big so we just
+        // pull in only the things we absolutely need.
+        datum.referenceAllele = refAllele;
+        datum.alternateAllele = altAllele;
+        dataManager.decodeAnnotations(datum, vc, true); //BUGBUG: when run with HierarchicalMicroScheduler this is
```

Comment:
"BUGBUG" comment is obsolete -- the `HierarchicalMicroScheduler` does not exist in GATK4.